build/mac/download_hermetic_xcode.py

```diff
@@ -0,0 +1,153 @@
+  return hermetic_xcode_version_plist['CFBundleShortVersionString']
+
+
+def InstallXcodeBinaries():
+  """Installs the Xcode binaries and accepts the license.
+  """
+  binaries_root = os.path.join(TOOLCHAIN_ROOT, 'xcode_binaries')
+  if (XCODE_VERSION == GetHermeticXcodeVersion(binaries_root) and not
+      os.path.islink(binaries_root)):
+    print(f"Hermetic Xcode {XCODE_VERSION} already installed")
+    return 0
+
+  url = HERMETIC_XCODE_BINARY
+  print(f"Downloading hermetic Xcode: {url}")
+  try:
+    deps.DownloadAndUnpack(url, binaries_root)
+  except URLError:
+      print(f"Failed to download hermetic Xcode: {url}")
+      print("Exiting.")
+      return 1
+
+  # Accept the license for this version of Xcode if it's newer than the
+  # currently accepted version.
+  hermetic_xcode_version = GetHermeticXcodeVersion(binaries_root)
+
+  hermetic_xcode_license_path = os.path.join(binaries_root,
+                                   'Contents/Resources/LicenseInfo.plist')
+  hermetic_xcode_license_plist = LoadPList(hermetic_xcode_license_path)
+  hermetic_xcode_license_version = hermetic_xcode_license_plist['licenseID']
+
+  should_overwrite_license = True
+  current_license_path = '/Library/Preferences/com.apple.dt.Xcode.plist'
+  if os.path.exists(current_license_path):
+    current_license_plist = LoadPList(current_license_path)
+    xcode_version = current_license_plist.get(
+        'IDEXcodeVersionForAgreedToGMLicense')
+    if (xcode_version is not None and pkg_resources.parse_version(xcode_version)
+        >= pkg_resources.parse_version(hermetic_xcode_version)):
+      should_overwrite_license = False
+
+  if not should_overwrite_license:
+    return 0
+
+  # Use puppet's sudoers script to accept the license if its available.
+  license_accept_script = '/usr/local/bin/xcode_accept_license.py'
+  if os.path.exists(license_accept_script):
+    args = [
+        'sudo', license_accept_script,
+        '--xcode-version', hermetic_xcode_version,
+        '--license-version', hermetic_xcode_license_version
+    ]
+    subprocess.check_call(args)
+    return 0
+
+  # Otherwise manually accept the license. This will prompt for sudo.
+  print('Accepting new Xcode license. Requires sudo.')
+  sys.stdout.flush()
+  args = [
+      'sudo', 'defaults', 'write', current_license_path,
+      'IDEXcodeVersionForAgreedToGMLicense', hermetic_xcode_version
+  ]
+  subprocess.check_call(args)
+  args = [
+      'sudo', 'defaults', 'write', current_license_path,
+      'IDELastGMLicenseAgreedTo', hermetic_xcode_license_version
+  ]
+  subprocess.check_call(args)
+  args = ['sudo', 'plutil', '-convert', 'xml1', current_license_path]
+  subprocess.check_call(args)
+
+  return 0
+
+
+def main():
+  parser = argparse.ArgumentParser(description='Download hermetic Xcode.')
+  parser.parse_args()
+
+  if not PlatformMeetsHermeticXcodeRequirements():
+    print('OS version does not support toolchain.')
```

Comment:
this should be more like an error/warning and contain words `Xcode` and `hermetic`.

Reply:
> this also comes directly from upstream, but I can clarify

I don't like to follow upstream blindly :) it's our script now, we can make it clean, simple and self-explanatory during a run.